### ISSUE TITLE
Rule Change: add `skipMath` option to `no-consecutive-blank-line`

### ISSUE THREAD

**Issue by lumirlumir** [eslint-markdown/eslint-markdown#688]
## Overview

Add a `skipMath` option to the [no-consecutive-blank-line](https://github.com/lumirlumir/npm-eslint-markdown/blob/main/packages/eslint-markdown/src/rules/no-consecutive-blank-line.ts) rule.

The rule scans source lines and can remove excess blank lines inside math blocks. It already supports excluding code blocks through `skipCode`; math blocks should have an equivalent option so their internal spacing can be managed separately from Markdown prose.

The option should be a boolean defaulting to `true` and apply only to parsed `math` nodes. Math parsing requires `languageOptions: { math: true }`. No `skipInlineMath` option is needed for this blank-line rule.

### 1. Update the Rule

In `packages/eslint-markdown/src/rules/no-consecutive-blank-line.ts`, add `skipMath: boolean` to `RuleOptions`, register it in `meta.schema`, and set its default to `true` in `meta.defaultOptions`. Read it in `create()` and add:

```ts
math(node) {
  if (skipMath) skipRanges.push(sourceCode.getRange(node));
},
```

Reuse the existing `SkipRanges` and per-line checking logic. Skipped lines must reset the consecutive blank-line count and receive neither reports nor autofixes. Preserve the existing `max`, `skipCode`, and EOF-fix behavior. Use AST ranges rather than matching math delimiters manually.

### 2. Add Tests

In `packages/eslint-markdown/src/rules/no-consecutive-blank-line.test.ts`, cover:

- With math parsing enabled, excess blank lines inside math blocks are skipped by default and with explicit `skipMath: true`.
- With `skipMath: false`, excess blank lines are reported and fixed according to `max`.
- Excess blank lines before and after math blocks are still checked; blank-line runs are not combined across an excluded block.
- `skipMath` and `skipCode` work independently.
- With math parsing disabled, math delimiters alone do not exclude blank lines.
- Fixes outside an excluded math block, including excess trailing blank lines at EOF, preserve its contents.

Use individually named tests and the existing helper to check both CommonMark and GFM.

### 3. Update the Documentation

Update `website/docs/rules/no-consecutive-blank-line.md` with the option's type, default, enabled/disabled examples, and the math-parsing prerequisite. Ensure automatically checked math examples enable math parsing.

Document that math blocks will be excluded by default when math parsing is enabled; `skipMath: false` preserves the previous behavior.

Related: #685, #686, #687.

## Validation

For a fresh clone, run `npm install` from the repository root before implementation or validation. After making the changes, run:

```sh
npm run test
npm run build
npm run lint
```

**Comment by hu6r1s** [comment]
I'd love to take a look at this issue and work on it as my first contribution. Is it okay if I give it a try?

**Comment by lumirlumir** [comment]
Thanks for your attention to this issue! A PR would be welcome.
